presentation/src/main/java/feature/compose/ComposeViewModel.kt

```diff
@@ -430,9 +436,25 @@ class ComposeViewModel(intent: Intent) : QkViewModel<ComposeView, ComposeState>(
                 .withLatestFrom(attachments, conversation, { body, attachments, conversation ->
                     val threadId = conversation.id
                     val addresses = conversation.recipients.map { it.address }
-                    sendMessage.execute(SendMessage.Params(threadId, addresses, body, attachments))
-                    view.setDraft("")
-                    this.attachments.onNext(ArrayList())
+
+                    // If we're not allowed to send SMS, ask for the permission.
+                    if (ContextCompat.checkSelfPermission(context, Manifest.permission.SEND_SMS)
+                            != PackageManager.PERMISSION_GRANTED) {
+                        ActivityCompat.requestPermissions(view as Activity,
+                                arrayOf(Manifest.permission.SEND_SMS),0)
+
+                    }
+
+                    // Ensure we were granted permission before sending anything
+                    if (ContextCompat.checkSelfPermission(context, Manifest.permission.SEND_SMS)
```

Comment:
This block will be executed immediately after the previous statement. The requestPermissions method acts asynchronously, and so this will execute before the user has a chance to respond to the dialog, rendering the second check useless